testing/test-setup/src/lib/extend/path.matcher.ts

```diff
@@ -0,0 +1,183 @@
+import type { SyncExpectationResult } from '@vitest/expect';
+import { expect } from 'vitest';
+import { osAgnosticPath } from '@code-pushup/test-utils';
+
+export type CustomPathMatchers = {
+  toMatchPath: (path: string) => void;
+  toStartWithPath: (path: string) => void;
+  toContainPath: (path: string) => void;
+  toEndWithPath: (path: string) => void;
+};
+
+export type CustomAsymmetricPathMatchers = {
+  /* eslint-disable @typescript-eslint/no-explicit-any */
+  pathToMatch: (path: string) => any;
+  pathToStartWith: (path: string) => any;
+  pathToContain: (path: string) => any;
+  pathToEndWith: (path: string) => any;
+  /* eslint-enable @typescript-eslint/no-explicit-any */
+};
+
+expect.extend({
+  toMatchPath(actual: string, expected: string): SyncExpectationResult {
+    const normalizedReceived = osAgnosticPath(actual);
+    const normalizedExpected = osAgnosticPath(expected);
+
+    const pass = normalizedReceived === normalizedExpected;
+    return pass
+      ? {
+          message: () => `expected ${actual} not to match path ${expected}`,
+          pass: true,
+          actual,
+          expected,
+        }
+      : {
+          message: () => `expected ${actual} to match path ${expected}`,
+          pass: false,
+          actual,
+          expected,
+        };
+  },
+
+  pathToMatch(actual: string, expected: string): SyncExpectationResult {
+    const normalizedReceived = osAgnosticPath(actual);
+    const normalizedExpected = osAgnosticPath(expected);
+
+    const pass = normalizedReceived === normalizedExpected;
+    return pass
+      ? {
+          message: () => `expected ${actual} not to match path ${expected}`,
+          pass: true,
+          actual,
+          expected,
+        }
+      : {
+          message: () => `expected ${actual} to match path ${expected}`,
+          pass: false,
+          actual,
+          expected,
+        };
+  },
```

Comment:
The implementation of `to...Path` and its `pathTo...` variant is always 100% the same. Why not reuse it?

```ts
expect.extends({
  toMatchPath: assertPathMatches,
  pathToMatch: assertPathMatches,

  toStartWithPath: assertPathStartsWith,
  pathToStartWith: assertPathStartsWith,

  toContainPath: assertPathContains,
  pathToContain: assertPathContains,

  toEndWithPath: assertPathEndsWith,
  pathToEndWith: assertPathEndsWith,
});

function assertPathMatches(actual: string, expected: string): SyncExpectationResult {
   // ...
}

function assertPathStartsWith(actual: string, expected: string): SyncExpectationResult {
   // ...
}

function assertPathContains(actual: string, expected: string): SyncExpectationResult {
   // ...
}

function assertPathEndsWith(actual: string, expected: string): SyncExpectationResult {
   // ...
}
```

Reply:
Really good point! Will definitely reuse it! 👍